packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts

```diff
@@ -196,14 +196,28 @@ export const createRdsLambda = (
  * add the name. We can figure out the right way to expose this to customers if needed, but for now we are not invoking `setResourceName`
  * because it would have no effect.
  */
-export const createLayerVersionCustomResource = (scope: Construct, resourceNames: SQLLambdaResourceNames): AwsCustomResource => {
+export const createLayerVersionCustomResource = (
+  scope: Construct,
+  resourceNames: SQLLambdaResourceNames,
+  context: TransformerContextProvider,
+): AwsCustomResource => {
   const { SQLLayerManifestBucket, SQLLayerManifestBucketRegion, SQLLayerVersionManifestKeyPrefix } = ResourceConstants.RESOURCES;
 
   const key = Fn.join('', [SQLLayerVersionManifestKeyPrefix, Fn.ref('AWS::Region')]);
 
   const manifestArn = `arn:aws:s3:::${SQLLayerManifestBucket}/${key}`;
 
   const resourceName = resourceNames.sqlLayerVersionResolverCustomResource;
+
+  // if deployment type is sandbox, use id in the format: resourceName-YYYY-MM-DD to avoid multiple deployments in the same day
+  // if deployment type is branch, use id in the original format
+  let physicalIdValue: string;
+  if (context?.deploymentIdentifier?.deploymentType === 'sandbox') {
```

Comment:
nit: consider a [TS type predicate](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates) to determine whether you have a sandbox type or not. That has the advantage of centralizing type introspection into a single function rather than scattering it throughout the project. It also allows TypeScript to infer the structure of the rest of the object even if the type structure gets hairy. See examples in `packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts` ([example](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts#L76-L87))